File: Scr_Java/modelo/TorreCanon.java
package modelo;

public class TorreCanon extends Torre {

    // Alto dano, poco alcance: dano 25, alcance 2, costo 90.
    public TorreCanon(Coordenada pos) {
        super(25, 2, 90, pos);
    }

    // Cada mejora suma +10 de dano y sube un nivel.
    @Override
    public void mejorar() {
        nivel++;
        dano += 10;
    }

    @Override
    public String getTipo() {
        return "Canon";
    }
}

File: cpp/modelo/TorreCanon.cpp
#include "TorreCanon.h"

// Stats fijos al nacer: alto danio, poco alcance: danio 25, alcance 2, costo 90.
TorreCanon::TorreCanon(const Coordenada& pos)
    : Torre(25, 2, 90, pos) {}

// Cada mejora suma +10 de danio (mas que el Arquero) y sube un nivel.
void TorreCanon::mejorar() {
    nivel++;
    danio += 10;
}

std::string TorreCanon::getTipo() const {
    return "Canon";
}